src/AI.java

```diff
@@ -1,66 +1,185 @@
-import java.util.LinkedList;
-import java.util.Queue;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.PriorityQueue;
 
-class coordinate {
-    int x, y, color, rowSize;
+class coordinate implements Comparable<coordinate> {
```

Comment:
Reminder to reformat to follow Java naming conventions later

Reply:
Ok I reopen that issue again.